packages/sirius-web/frontend/sirius-web-application/src/extension/ReferenceWidgetDocumentTransform.ts

```diff
@@ -12,26 +12,20 @@
  *******************************************************************************/
 
 import { DocumentTransform } from '@apollo/client';
-import { DocumentNode, FieldNode, InlineFragmentNode, Kind, SelectionNode, visit, FragmentSpreadNode } from 'graphql';
+import { DocumentNode, FragmentDefinitionNode, InlineFragmentNode, Kind, SelectionNode, visit } from 'graphql';
 
 const shouldTransform = (document: DocumentNode) => {
-  return (
-    document.definitions[0] &&
-    document.definitions[0].kind === Kind.OPERATION_DEFINITION &&
-    (document.definitions[0].name?.value === 'detailsEvent' || document.definitions[0].name?.value === 'formEvent')
+  return document.definitions.some(
+    (definition) =>
+      definition.kind === Kind.OPERATION_DEFINITION &&
+      (definition.name?.value === 'detailsEvent' ||
+        definition.name?.value === 'formEvent' ||
+        definition.name?.value === 'formDescriptionEditorEvent')
```

Comment:
Could we test instead that we have a form? Because this seems to indicate that the reference widget cannot work in the related elements, representations and diagram filters subscription and in any form based subscription created by anyone outside of Sirius Web.

It's better than before but I'm not fan of having such an important widget that can only work in three specific subscriptions...

Reply:
The only other solution I can think of is to parse all selectionSets to check for a `FormRefreshedEventPayload`
Something like :
`const stack = operationDefinition.selectionSet.selections;
        while (stack.length > 0) {
            const selection = stack.pop();
            if (selection.kind === 'InlineFragment' &&
                selection.typeCondition.kind === 'NamedType' &&
                selection.typeCondition.name.value === 'FormRefreshedEventPayload') {
                return true;
            } else if (selection.selectionSet) {
                stack.push(...selection.selectionSet.selections);
            }
        }`
don't know if it's a better option...